prost/src/encoding.rs

```diff
@@ -721,6 +721,128 @@ fixed_width!(
     get_i64_le
 );
 
+pub mod enumeration {
+    use super::*;
+    use crate::OpenEnum;
+
+    pub fn encode<T, B>(tag: u32, value: &OpenEnum<T>, buf: &mut B)
+    where
+        T: Clone + Into<i32>,
+        B: BufMut,
+    {
+        int32::encode(tag, &value.to_raw(), buf)
+    }
+
+    pub fn merge<T, B>(
+        wire_type: WireType,
+        value: &mut OpenEnum<T>,
+        buf: &mut B,
+        _ctx: DecodeContext,
+    ) -> Result<(), DecodeError>
+    where
+        i32: TryInto<T>,
+        B: Buf,
+    {
+        let mut raw = 0;
+        int32::merge(wire_type, &mut raw, buf, DecodeContext::default())?;
+        *value = OpenEnum::from_raw(raw);
+        Ok(())
+    }
+
+    pub fn encode_repeated<T, B>(tag: u32, values: &[OpenEnum<T>], buf: &mut B)
+    where
+        T: Clone + Into<i32>,
+        B: BufMut,
+    {
+        for value in values {
+            encode(tag, value, buf);
+        }
```

Comment:
Could `int32::encode_repeated` be reused here?

Reply:
No, unless `int32::encode_repeated` is generalized to take an `impl IntoIterator` that produces `i32`. Which might be a good idea.

Reply:
So that would need a breaking change for `encode_repeated` which I don't think is worth the little code duplication at this point.